discussion/avoiding-mutable-aliases-to-hardware.tex

```diff
@@ -85,25 +88,24 @@ \subsection{Alternative Approaches}
+This is because the part of the program that is currently executing is put on hold in favor of the interrupt handler, and the new execution scope defines a different program environment.
 This makes it impossible to keep program state consistent between the various handlers unless it is kept in global variables.
-There are of course other approaches to this problem than to rely on the compiler to prevent the data races.
+There are, of course, other approaches to this problem other than to rely on the compiler to prevent the data races.
 
-One way to try and \emph{prevent} this problem during or after implementation can be to \emph{test} the program for errors that happens due to simultaneously occurring interrupts.
+One way to try and \emph{prevent} this problem during or after implementation can be to \emph{test} the program for errors that happen due to simultaneously occurring interrupts.
 There are several different approaches to how this can be done \cite{Higashi2010,Regehr2005}, but this has not been in the scope of this thesis.
-% but there is nothing special with {\rust} that makes the language more helpful than others for this purpose.
 
 Another approach can be to treat this as a concurrency problem that we can try to solve \emph{dynamically} instead of \emph{statically}.
 We can consider the problem that we have described throughout this section to be an application-dependent issue.
 Because of this, we can argue that it is not the right decision to solve this issue as part a low-level library like {\emlib} and its bindings.
 Instead, we can define an auxiliary library that can be implemented to handle the problem in an opinionated fashion, which might apply to a lot of different applications, but not to all.
 
-In \autoref{sec:irq-closures} we described the implementation of a library that takes on the task of hiding the various interrupt handlers and instead expose interrupts through \emph{closures} that can be registered at runtime.
+In \autoref{sec:irq-closures}, we described the implementation of a library that takes on the task of hiding the various interrupt handlers and instead expose interrupts through \emph{closures} that can be registered at runtime.
 It is possible to extend this library to make it \emph{safe} to dispatch new interrupts at runtime.
 E.g. by identifying the different possibilities of states the program can be in at the time of an interrupt, as well as the various actions that can be done depending on this state.
 The library can then provide routines to either \emph{lock} the access to a peripheral during an interrupt or provide \emph{channels} that can be used to communicate directly with the peripheral in a safe manner.
 Note that this is not an issue that can only be solved with {\rust} (it can be implemented in any language), but {\rust} might prove to be exceptionally well-suited for the task.
```

Comment:
Think this is better: `Note that this is not an issue that can only be solved with {\rust}, it can be implemented in any language, but {\rust} might prove to be exceptionally well-suited for the task.`